### PR TITLE
Feature/de sharedresource doc update

### PR DESCRIPTION
**Pull request by satenderrathee**
📲 What
Data Platform documentation update.

🤔 Why
6720

🛠 How
md files

👀 Evidence
npm run build & npm run test

🕵️ How to test
npm run build & npm run test

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+id: de_shared_resourcedeployment_azure
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:2]: Maybe rename this id (and file name) to `shared_resources_deployment_azure`?

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+title: DE shared resource Deployment
+sidebar_label: DE shared resource Deployment
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:4]:
```suggestion
title: Shared Resources Deployment
sidebar_label: Shared Resources Deployment
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+hide_title: false
+hide_table_of_contents: false
+description: DE shared resource deployment
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:7]:
```suggestion
description: Deployment of common Azure Data Factory resources shared by data pipelines
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+It assumes all [prerequisites](../requirements_data_azure.md#azure) are in place, including:
+
+* Azure subscription and service principal
+* Azure DevOps project with [Pipelines variable groups](../requirements_data_azure.md#azure-pipelines-variable-groups)
+* A [deployed Stacks data platform](core_data_platform_deployment_azure.md)
+
+## Step 1: Create feature branch
+
+Open the project locally and create a new feature branch e.g.:
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:28]:
```suggestion
Open the project locally and create a new feature branch, e.g.:
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+sidebar_label: DE shared resource Deployment
+hide_title: false
+hide_table_of_contents: false
+description: DE shared resource deployment
+keywords:
+  - DE
+  - data
+  - infrastructure
+  - azure
+  - template
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:13]:
```suggestion
  - adf
  - cicd
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+---
+
+This section provides an overview of deploying Data Platform shared resources for Stacks Data Platform.
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:16]:
```suggestion
This section provides an overview of deploying the shared resources for Stacks Data Platform.
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+* A [deployed Stacks data platform](core_data_platform_deployment_azure.md)
+
+## Step 1: Create feature branch
+
+Open the project locally and create a new feature branch e.g.:
+
+```bash
+git checkout -b feat/de-shared-pipeline
+```
+The `de_build` folder includes YAML file called `job-pipeline-vars` that contains the variables used in the DE shared reource pipeline. These variables must be updated as per the project requirements.
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:33]: Please run `/azp run` for formatting checks

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+## Step 2: Add DE shared resource pipeline in Azure DevOps
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:35]:
```suggestion
## Step 2: Add a Shared Resources pipeline in Azure DevOps
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+YAML file containing a template Azure DevOps CI/CD pipeline for building and deploying the DE shared reource, named `de-shared-resources.yml` is configured in the `de_workloads/shared_resources`
+
+YAML file `de-shared-resources.yml` should be added as the definition for a new pipeline in Azure DevOps.
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:39]:
```suggestion
YAML file containing a template Azure DevOps CI/CD pipeline for building and deploying the shared resources, named `de-shared-resources.yml` is configured in the [de_workloads/shared_resources](https://github.com/amido/stacks-azure-data/tree/main/de_workloads/shared_resources).
This YAML file should be added as the definition for a new pipeline in Azure DevOps.
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+1. Sign-in to your Azure DevOps organization and go to your project
+2. Go to Pipelines, and then select **New pipeline**
+3. Name the new pipeline, e.g. `de-shared-resources`
+4. For the pipeline definition, specify the YAML file in the project repository feature branch (`de-shared-resources.yml`) and save
+5. The new pipeline will require access to any Azure DevOps pipeline variable groups specified in the pipeline YMAL. Under each variable group, go to 'Pipeline permissions' and add the pipeline.
+
+
+## Step 3: Deploy DE shared reource in non-production environment
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:48]:
```suggestion
## Step 3: Deploy shared resources in non-production environment
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+Run the pipeline configured in Step 2 to commence the build and deployment process.
+
+Running this pipeline in Azure DevOps will initiate the deployment of artifacts into the non-production (nonprod) environment. It's important to monitor the progress of this deployment to ensure its success. You can track the progress and status of the deployment within the Pipelines section of Azure DevOps.
+
+If successful, the core DE shared reource will now be available in the nonprod Stacks environment. To view these deployed resources, navigate to the [Azure portal](https://portal.azure.com/) and search for the resource group and ope ADF. Within the ADF, you'll find the shared resources that were deployed.
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:54]:
```suggestion
If successful, the core DE shared reource will now be available in the nonprod Stacks environment. To view these deployed resources, navigate to the [Azure portal](https://portal.azure.com/), and go to your Azure Data Factory UI in the proper resource group. Within the ADF, you'll find the shared resources that were deployed.
```

Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:54]: Maybe some screenshot of these shared resources in ADF? I think we should somehow mention what exactly are those.

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+## Step 4: Deploy DE shared reource in further environments
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:56]:
```suggestion
## Step 4: Deploy shared resources in further environments
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+By default Stacks provides a framework for managing the platform across two environments - nonprod and prod.
+The template CI/CD pipelines provided are based upon these two platform environments (nonprod and prod) - but these may be amended depending upon the specific requirements of your project and organisation.
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:59]:
```suggestion
The template CI/CD pipelines provided are based upon these two platform environments, but these may be amended depending upon the specific requirements of your project and organisation.
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
+---
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md:1]: General remark: please make sure that docs in stacks-azure-data are consistent with these (make sure that you run  pre-commit hooks in that repo to check/fix linting)

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md (hunk 1 above)
Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md]: updated

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md (hunk 11 above)
Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md]: added resource list

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+  - DE
+  - data
+  - infrastructure
+  - adf
+  - cicd
+---
+
+This section provides an overview of deploying the shared resources for Stacks Data Platform.
+This aligns to the workflow shown in the [deployment architecture](../architecture/architecture_data_azure.md#data-engineering-workloads) section.
+Data Plaform shared resources include the ADF resources which are shared in various ADF pipelines.These are as follow:-
```
Comment (motylp) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:18]:
```suggestion
Data Plaform shared resources include the ADF resources which are shared in various ADF pipelines.These are as follows:
```

File: docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md (hunk 14 above)
Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/de_shared_resourcedeployment_azure.md]: yeah sure

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+  - DE
+  - data
+  - infrastructure
+  - adf
+  - cicd
+---
+
+This section provides an overview of deploying the shared resources for Stacks Data Platform.
+This aligns to the workflow shown in the [deployment architecture](../architecture/architecture_data_azure.md#data-engineering-workloads) section.
+Data Plaform shared resources include the ADF resources which are shared in various ADF pipelines.These are as follows:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:18]:
```suggestion
The shared resources include Azure Data Factory resources which are shared across pipelines. These are as follows:
```

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    * ls_KeyVault
+    * ls_ADLS_DataLake
+    * ls_Databricks_Small
+    * ls_Blob_ConfigStore
+* Datasets
+    * ds_dp_ConfigStore_Json
+    * ds_dp_DataLake_Parquet
+* Pipelines
+    * pipeline_Get_Ingest_Config
+    * pipeline_Generate_Ingest_Query
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:31]:
```suggestion
* Linked services
    * _ls_ADLS_DataLake_ - Connection to the Data Lake
    * _ls_Blob_ConfigStore_ - Connection to the config storage location
    * _ls_Databricks_Small_ - Connection to Databricks job cluster (default 2 fixed workers)
    * _ls_KeyVault_ - Connection to Azure Key Vault
* Datasets
    * _ds_dp_ConfigStore_Json_ - For reading JSON data from ls_Blob_ConfigStore
    * _ds_dp_DataLake_Parquet_ - For writing Parquet data to ls_ADLS_DataLake
* Pipelines
    * _pipeline_Get_Ingest_Config_ - To retrieve config data for use in a pipeline
    * _pipeline_Generate_Ingest_Query_ - To generate a query for ingesting data

For details of how these resources are used in ingest pipelines, see [data ingestion](../etl_pipelines/ingest_data_azure.md).
```

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+It assumes all [prerequisites](../requirements_data_azure.md#azure) are in place, including:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:33]:
```suggestion
This guide assumes all [prerequisites](../requirements_data_azure.md#azure) are in place, including:
```

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+## Step 2: Add a shared resources pipeline in Azure DevOps
+
+YAML file containing a template Azure DevOps CI/CD pipeline for building and deploying the shared resources, named `de-shared-resources.yml` is configured in the [de_workloads/shared_resources](https://github.com/amido/stacks-azure-data/tree/main/de_workloads/shared_resources).
+This YAML file should be added as the definition for a new pipeline in Azure DevOps.
+
+1. Sign-in to your Azure DevOps organization and go to your project
+2. Go to Pipelines, and then select **New pipeline**
+3. Name the new pipeline, e.g. `de-shared-resources`
+4. For the pipeline definition, specify the YAML file in the project repository feature branch (`de-shared-resources.yml`) and save
+5. The new pipeline will require access to any Azure DevOps pipeline variable groups specified in the pipeline YMAL. Under each variable group, go to 'Pipeline permissions' and add the pipeline.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:58]:
```suggestion
5. The new pipeline will require access to any Azure DevOps pipeline variable groups specified in the pipeline YAML. Under each variable group, go to 'Pipeline permissions' and add the pipeline.
```

File: docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+
+## Step 3: Deploy shared resources in non-production environment
+
+Run the pipeline configured in Step 2 to commence the build and deployment process.
+
+Running this pipeline in Azure DevOps will initiate the deployment of artifacts into the non-production (nonprod) environment. It's important to monitor the progress of this deployment to ensure its success. You can track the progress and status of the deployment within the Pipelines section of Azure DevOps.
+
+If successful, the core DE shared reource will now be available in the nonprod Stacks environment. To view these deployed resources, navigate to the [Azure portal](https://portal.azure.com/), and go to your Azure Data Factory UI in the proper resource group. Within the ADF, you'll find the shared resources that were deployed.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/shared_resources_deployment_azure.md:67]:
```suggestion
If successful, the core DE shared reources will now be available in the nonprod Stacks environment. To view the deployed resources, navigate to the relevant resource group in the [Azure portal](https://portal.azure.com/). The deployed Data Factory resources can be viewed through the [Data Factory UI](https://adf.azure.com/).
```

**Comment by adurkan-amido** [comment]
Can we also add an extra bullet point to the pre-requisites on the `getting_started/etl_pipelines_deployment_azure.md` page, that links back to this.

**Comment by satenderrathee** [comment]
> Can we also add an extra bullet point to the pre-requisites on the `getting_started/etl_pipelines_deployment_azure.md` page, that links back to this.

updated

File: docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md [hunk 21]
```diff
@@ -21,6 +21,7 @@ This aligns to the workflow shown in the [deployment architecture](../architectu
 It assumes all prerequisites are in place, including:
 
 * A [deployed Stacks data platform](core_data_platform_deployment_azure.md)
+* A [deployed shared resources for Stacks data platform](shared_resources_deployment_azure.md)
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md:24]:
```suggestion
* [Deployed shared resources](shared_resources_deployment_azure.md)
```

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**